core/src/main/java/dk/alexandra/fresco/framework/util/StrictBitVector.java

```diff
@@ -0,0 +1,219 @@
+   * @param isBigEndian
+   *          Indicates whether the underlying byte array should be interpreted
+   *          as big-endian or little-endian
+   */
+  public void setBit(int index, boolean value, boolean isBigEndian) {
+    rangeCheck(index);
+    int actualIndex = isBigEndian ? size - 1 - index : index;
+    ByteArrayHelper.setBit(bits, actualIndex, value);
+  }
+
+  @Override
+  public void setBit(int index, boolean value) {
+    setBit(index, value, true);
+  }
+
+  @Override
+  public int getSize() {
+    return size;
+  }
+
+  /**
+   * Returns a reference to the internal byte array representing the bit vector.
+   * Thus modifying this  byte array DIRECTLY modifies the bits in this
+   * StrictBitVector object.
+   * <p>
+   * The representation is big-endian, that is the first 8 bits will be in the
+   * last byte.
+   * </p>
+   *
+   * @return A byte array with the content of this bit vector
+   */
+  @Override
+  public byte[] toByteArray() {
+    return bits;
+  }
+
+  /**
+   * Generates a string representation of this vector.
+   *
+   * @return A string with the binary representation of this vector
+   */
+  public String asBinaryString() {
+    String binStr = "";
+    for (int b = 0; b < size; b++) {
+      binStr += getBit(b) ? "1" : "0";
+    }
+    return binStr;
+  }
+
+  @Override
+  public void xor(BitVector other) {
+    ByteArrayHelper.xor(bits, other.toByteArray());
+  }
+
+  @Override
+  public String toString() {
+    return "StrictBitVector [bits=" + Arrays.toString(bits) + "]";
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + Arrays.hashCode(bits);
+    // no need to include size parameter since it is derived directly from bits
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (getClass() != obj.getClass()) {
+      return false;
+    }
+    StrictBitVector other = (StrictBitVector) obj;
+    if (!Arrays.equals(bits, other.bits)) {
+      return false;
+    }
+    // no need to check size parameter since it is derived directly from bits
+    return true;
+  }
+
+  /**
+   * Constructs a new StrictBitVector which is the concatenation of the vectors
+   * given as input.
+   *
+   * @param bitVectors
+   *          the vectors to concatenate
+   * @return the concatenated vector
+   */
+  public static StrictBitVector concat(StrictBitVector... bitVectors) {
+    return concat(false, bitVectors);
+  }
+
+  /**
+   * Constructs a new StrictBitVector which is the concatenation of the vectors
+   * given as input.
+   *
+   * @param reverse
+   *          indicating whether or not to reverse the list before concatenation
+   * @param bitVectors
+   *          the vectors to concatenate
+   * @return the concatenated vector
+   */
+  public static StrictBitVector concat(boolean reverse,
```

Comment:
Reverse is an odd logic - it is only used one place. I propose to move this to the point of call rather than here.

Reply:
Removed in commit dddb1730